### DISCUSSION TITLE
now dev with path segment directory

### DISCUSSION THREAD

**Discussion by dougwt** [vercel/vercel#4170] [General]
Hello, I am trying to configure my api routes using [path segments](https://vercel.com/docs/v2/serverless-functions/introduction/#path-segments) in the form of `api/collections/[collection_id]/` with the following `now.json` and `api/` directory structure:

```
    {
      "env": {
        "NODE_ENV": "production",
    		...
      },
      "trailingSlash": true
    }
```
```
    api
    └── collections
        ├── [collection_id]
        │   └── index.js
        └── index.js
```

Everything is working properly when deployed to production, however I have trouble reaching my endpoints when using `now dev` . 

`http GET http://localhost:3000/api/collections/5ea53b7ef580eb0007828634`

properly redirects to the trailing slash version, as I have configured, however when I try to request the path with the trailing slash, it returns my React SPA. The same is also true when I add index.js to the end of the path.

`http GET [http://localhost:3000/api/collections/5ea53b7ef580eb0007828634/`

`http GET [http://localhost:3000/api/collections/5ea53b7ef580eb0007828634/index.js`

For comparison, here is the correctly working production endpoint: https://bujoy-git-vercel-refactor.dougwt.now.sh/api/collections/5ea53b7ef580eb0007828634/

The full source for my deployment is available at https://vercel.com/dougwt/bujoy/f509qm9vd/source

Is this a known issue or am I doing something wrong? Any guidance would be greatly appreciated!

**Comment by TooTallNate** [comment]
Hey there @dougwt, thanks for the question. I tried to reproduce your issue but so far cannot. I created an [example repository](https://github.com/TooTallNate/vercel-discussion-4170) that works correctly as far as I can tell. To wit:

```
$ curl localhost:3000/api
Redirecting to /api/ (308)

$ curl localhost:3000/api/
{"pid":67877,"query":{}}

$ curl localhost:3000/api/test
Redirecting to /api/test/ (308)

$ curl localhost:3000/api/test/
{"url":"/api/test/?segment=test","query":{"segment":"test"}}
```

Let me know if I am missing something!

**Reply by dougwt** [reply]
Hello, I really appreciate you taking a look at my issue! I was able to confirm that your example worked properly on now dev with my local machine.

I have created a pull request with the addition of my own package.json and src/ directory containing my React client and the issue is reproduced. Perhaps it is something in my package.json?

Edit: I have concluded that the package.json is causing the issue, because I am able to remove it from my original project source and the dev server api paths work as expected.

Edit2: My issue appears to be with using parcel in my npm dev and build scripts. When I edit them to be empty strings, everything works as it should.

```
"scripts": {
    "dev": "parcel src/index.html -d public",
    "build": "parcel build src/index.html -d public"
  },
```

Is there something about the way parcel works that is incompatible?
